test/integration/targets/lookups/tasks/main.yml

```diff
@@ -154,6 +154,26 @@
     that:
         - "test_val == home_var_value.stdout"
 
+# UNICODE LOOKUP
+
+# https://github.com/ansible/ansible/issues/65297
+- name: set variable that has unicode characters
+  set_fact:
+      UNICODE_VAR: "𝖆𝖓𝖘𝖎𝖇𝖑𝖊"
+
+- name: confirm env lookup works with unicode values
+  shell: "echo $UNICODE_VAR"
+  register: unicode_var_value
```

Comment:
I think this test case is not correct. Since `set_fact` does not set environment variable in system. The second shell task echoes empty `$UNICODE_VAR` and registering it in `unicode_var_value` and last task is just checking `test_unicode_val == unicode_var_value.stdout` which is true (evaluates as `'' == ''`).

Reply:
```
TASK [set variable that has unicode characters] **************************************************************
ok: [127.0.0.1] => {"ansible_facts": {"UNICODE_VAR": "𝖆𝖓𝖘𝖎𝖇𝖑𝖊"}, "changed": false}

TASK [confirm env lookup works with unicode values] **********************************************************
changed: [127.0.0.1] => {"changed": true, "cmd": "echo $UNICODE_VAR", "delta": "0:00:00.004343", "end": "2019-12-21 08:58:19.570958", "rc": 0, "start": "2019-12-21 08:58:19.566615", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [use env lookup to get UNICODE_VAR value] ***************************************************************
ok: [127.0.0.1] => {"ansible_facts": {"test_unicode_val": ""}, "changed": false}

TASK [compare unicode values] ********************************************************************************
ok: [127.0.0.1] => {
    "changed": false,
    "msg": "All assertions passed"
}
```
See `test_unicode_val` and `unicode_var_value.stdout`

Reply:
i changed the code to following:

```yml
# https://github.com/ansible/ansible/issues/65297
- name: set variable that has unicode characters
  set_fact:
    UNICODE_VAR: "𝖆𝖓𝖘𝖎𝖇𝖑𝖊"

- name: confirm env lookup works with unicode values
  shell: "echo $UNICODE_VAR"
  register: unicode_var_value
  environment:
    UNICODE_VAR: "𝖆𝖓𝖘𝖎𝖇𝖑𝖊"

- name: use env lookup to get UNICODE_VAR value
  set_fact:
    test_unicode_val: "{{ lookup('env', 'UNICODE_VAR') }}"
  environment:
    UNICODE_VAR: "𝖆𝖓𝖘𝖎𝖇𝖑𝖊"

- debug: var=unicode_var_value
- debug: var=test_unicode_val

- name: compare unicode values
  assert:
    that:
      - "test_unicode_val == unicode_var_value.stdout"
```

though, the `unicode_var_value` gets the correct value for `UNICODE_VAR`, `test_unicode_val` doesn't:

```bash

TASK [lookups : set variable that has unicode characters] **********************
ok: [localhost]

TASK [lookups : confirm env lookup works with unicode values] ******************
changed: [localhost]

TASK [lookups : use env lookup to get UNICODE_VAR value] ***********************
ok: [localhost]

TASK [lookups : debug] *********************************************************
ok: [localhost] => {
    "unicode_var_value": {
        "changed": true,
        "cmd": "echo $UNICODE_VAR",
        "delta": "0:00:00.003674",
        "end": "2019-12-21 04:48:43.307554",
        "failed": false,
        "rc": 0,
        "start": "2019-12-21 04:48:43.303880",
        "stderr": "",
        "stderr_lines": [],
        "stdout": "𝖆𝖓𝖘𝖎𝖇𝖑𝖊",
        "stdout_lines": [
            "ansible"
        ]
    }
}

TASK [lookups : debug] *********************************************************
ok: [localhost] => {
    "test_unicode_val": ""
}

TASK [lookups : compare unicode values] ****************************************
fatal: [localhost]: FAILED! => {
    "assertion": "test_unicode_val == unicode_var_value.stdout",
    "changed": false,
    "evaluated_to": false,
    "msg": "Assertion failed"
}
```

Reply:
@abhaykadam Could you please try adding `export UNICODE_VAR=𝖆𝖓𝖘𝖎𝖇𝖑𝖊` in `runme.sh` located in ` test/integration/targets/lookups/` and re-run?

Reply:
😂 was just testing that... have pushed the changes... and it's working!

Reply:
```bash
TASK [lookups : get UNICODE_VAR environment var value] *************************                                                                                                     
changed: [localhost]                                                                                                                                                                 
                                                                                          
TASK [lookups : use env lookup to get UNICODE_VAR value] ***********************          
ok: [localhost]                              
                                                                                          
TASK [lookups : debug] *********************************************************          
ok: [localhost] => {                                                                                                                                                                 
    "unicode_var_value": {        
        "changed": true,                     
        "cmd": "echo $UNICODE_VAR",
        "delta": "0:00:00.005369",
        "end": "2019-12-21 05:41:44.396259",
        "failed": false,
        "rc": 0,
        "start": "2019-12-21 05:41:44.390890",
        "stderr": "",
        "stderr_lines": [],
        "stdout": "café",
        "stdout_lines": [
            "café"
        ]
    }
}

TASK [lookups : debug] *********************************************************
ok: [localhost] => {
    "test_unicode_val": "café"
}

TASK [lookups : compare unicode values] ****************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}
```